crates/types/src/protobuf.rs

```diff
@@ -89,13 +109,21 @@ pub mod node {
             msg_id: u64,
             in_response_to: Option<u64>,
         ) -> Self {
+            let mut span_context = SpanContext::default();
+            let context = tracing::Span::current().context();
+
+            global::get_text_map_propagator(|propegator| {
+                propegator.inject_context(&context, &mut span_context)
+            });
```

Comment:
Are we also propagating the span information if we are not sampling this span?

Reply:
Thanks for pointing this out. I changed it so that we only propagate it iff the context span is sampled.